packages/clay-link/src/index.tsx

```diff
@@ -37,19 +38,25 @@ const ClayLink: React.FunctionComponent<IProps> = ({
 	monospaced,
 	outline,
 	...otherProps
-}: IProps) => (
-	<a
-		className={classNames(className, {
-			'link-monospaced': monospaced,
-			'link-outline': outline,
-			'link-outline-borderless': borderless,
-			[`link-${displayType}`]: displayType && !outline,
-			[`link-outline-${displayType}`]: displayType && outline,
-		})}
-		{...otherProps}
-	>
-		{children}
-	</a>
-);
+}: IProps) => {
+	const TagOrComponent = useContext(ClayLinkContext);
+
+	return (
+		<TagOrComponent
+			className={classNames(className, {
+				'link-monospaced': monospaced,
+				'link-outline': outline,
+				'link-outline-borderless': borderless,
+				[`link-${displayType}`]: displayType && !outline,
+				[`link-outline-${displayType}`]: displayType && outline,
+			})}
+			{...otherProps}
+		>
+			{children}
+		</TagOrComponent>
+	);
+};
+
+export {ClayLinkContext};
```

Comment:
hey @bryceosterhaus yet i'm not sure about that, i wish we could test this out, so i suggest we rename it to unstable (being experimental) at the moment and as soon as we feel and collect feedback that this is the right direction. We can make it stable.

```suggestion
export {ClayLinkContext as unstable_ClayLinkContext};
```

I think it's safer that way before it makes it stable along with the official release.

Reply:
You can test it out in the storybook and also see it working in the tests. Or are you thinking a different way to test it?

Reply:
Also, it looks like it was also brought up here by Diego yesterday, https://github.com/liferay/clay/issues/2280. Not sure if this overlaps. Ill go ahead and add a commit for  forwardRef as well

Reply:
> You can test it out in the storybook and also see it working in the tests. Or are you thinking a different way to test it?

Hey @bryceosterhaus wouldn't really be these tests, what I mean by that is we need to test with the teams and wait a while to know if this approach is the best way to get on with it.

Another concern that I would like to test in a real application is that we might want `ClayLink` components without being attached to a `react-router` for example, so we need to test how this will get inside those applications, add it as `unstable` gives us the ability to modify this at any time and test with teams even when we make the official release.

> Also, it looks like it was also brought up here by Diego yesterday, #2280. Not sure if this overlaps. Ill go ahead and add a commit for forwardRef as well

Yeah this is another use case we just found in DXP Cloud (In the new admin) using `next/router`, you can pass as children (`ClayLink`) on `next/router` for example that it will replace `ClayLink` markup but it adds a ref to the element.